gate.py

```
@@ -0,0 +1,348 @@
+    get_current_user,
+)
+from app.crypto import encrypt_answers, generate_csrf_token, generate_token, shuffle_questions
+from app.db import get_pool
+from app.middleware import CSRF_COOKIE, CSRF_FIELD, set_csrf_cookie
+from app.questions import GATE_QUESTIONS, MAIN_QUESTIONS, QUESTIONS, TOTAL_QUESTIONS
+
+router = APIRouter()
+templates = Jinja2Templates(directory="app/templates")
+
+
+def _require_user(user: dict | None):
```

Ruff (0.15.9)

[warning] Missing return type annotation for private function `_require_user`

(ANN202)